### DISCUSSION TITLE
[2024-07-22] GUI Team Demo Day

### DISCUSSION THREAD

**Discussion by farmaazon** [enso-org/enso#10623] [Weekly Demo Day]
* #10457 - putting just any code resulted in node being broken

[Screencast from 2024-07-22 15-15-04.webm](https://github.com/user-attachments/assets/1d08a470-dae6-44d4-b68c-b34933d775dd)

* #10509 - also removed unnecessary setExecutionEnvironment call on startup

[Screencast from 2024-07-10 13-55-46.webm](https://github.com/enso-org/enso/assets/3919101/4ad7c039-84f6-4e42-aad9-5e287ccd88bb)

* #10546

![image](https://github.com/user-attachments/assets/21efe4e0-922b-4a83-8f6e-ddaea360e655)

* #10565

* #10577 

[Screencast from 2024-07-17 15-18-28.webm](https://github.com/user-attachments/assets/91d48759-3d44-47ac-bbd2-b9a6085ade82)

* #10607

![image](https://github.com/user-attachments/assets/308cbe03-3bcb-48d6-a4c6-899469c5a2f4)

* #10542 - my first contribution Language Server code: protocol initialization becomes idempotent.

**Comment by vitvakatu** [comment]
- https://github.com/enso-org/enso/pull/10506
- https://github.com/enso-org/enso/pull/10523
- https://github.com/enso-org/enso/issues/10435
   - https://github.com/enso-org/enso/pull/10558
  - https://github.com/enso-org/enso/pull/10590
  - https://github.com/enso-org/enso/pull/10591

**Comment by kazcw** [comment]
### Dashboard/IDE integration + UI

- #10400
- #10167

https://github.com/user-attachments/assets/54cc375d-2a77-431c-af85-75fc872e2bcc

### Enso Language

- #10516
- #10366

**Comment by Frizi** [comment]
- Fixed issues with widgets losing focus after edits by changing semantics of our internal Node IDs https://github.com/enso-org/enso/pull/10512

https://github.com/enso-org/enso/assets/919491/72d9ec29-54ed-4899-aac1-6678a001a21d

- Added support for arbitrarily-shaped widgets. Refactored handling of margins between widgets, which improved looks of nodes ending with a port. https://github.com/enso-org/enso/pull/10599

<img width="371" alt="image" src="https://github.com/user-attachments/assets/4aa2ae0f-06f6-45fd-b7a1-1b7c0f60d395">

**Comment by JaroslavTulach** [comment]
Engine team updates:
- #10579 had a fix when backend takes long to open a file but decided that GUI should fix it
    - reported as #10606
- #10481 - additional `projectsDirectory` parameter for Project Manager
-  #10572 - Project Manager now accepts configs with no namespace
- `Any.to` method support https://github.com/enso-org/enso/issues/10545#issuecomment-2225867179 - needs to be done in the IDE
- Working on #10440
- VSCode shows error: #10553


![underlined errors](https://github.com/user-attachments/assets/4e5df6db-d2a9-41ff-b54e-496f5cc888c4)
